### DISCUSSION TITLE
Why does my product look different depending on the ambient light?

### DISCUSSION THREAD

**Discussion by bysmoke** [google/model-viewer#4722] [Q&A]
I am adding two examples. My product changes color depending on the light of the room with my movements. How come it is almost not visible in the dark environment, but the one in the other video is not affected at all? I am sharing my own code.

https://github.com/google/model-viewer/assets/162462298/63c9a1f8-7dd8-40f3-af32-49c578785afa

https://github.com/google/model-viewer/assets/162462298/bb89bf42-e110-4f17-874e-308b811dde9d

[soru.txt](https://github.com/google/model-viewer/files/14727294/soru.txt)

**Comment by elalish** [comment]
So, the difference you're seeing is that the top video is using WebXR mode and the bottom is using Scene-Viewer (an external app). Lighting estimation is enabled in Scene-Viewer, but disabled by default in WebXR (because it sometimes doesn't work well) - you can enable it with this to get similar results: https://modelviewer.dev/docs/#entrydocs-augmentedreality-attributes-xrEnvironment

I would actually say the dark chair in your second video looks more realistic to me, which is the purpose of lighting estimation, but true it also gets overly dark at times. The AI is far from perfect.

**Reply by bysmoke** [reply]
the dark model is more stable control, the correct interaction, the better model, the full scale stands Decently in place, the light model sometimes surprises the control, as I understand, it is unlikely to do it all together. thanks for your answer

**Comment by bysmoke** [comment]
can you give me a code example so that I can make a bright experiment for sceneviewer, I can play games on the code and maybe try it like that, I can chat with gemini and print only so much code :)

    ar-modes="webxr scene-viewer quick-look"                    ar-modes="scene-viewer"        is this change useful

**Reply by elalish** [reply]
Sorry, there's no customization for Scene Viewer - it's a separate app.
